classes/webservice/WebserviceSpecificManagementAttachments.php

```diff
@@ -0,0 +1,300 @@
+                    $attachment = new Attachment((int) $this->wsObject->urlSegment[1]);
+                    $attachment->delete();
+                    break;
+            }
+        } else {
+            // Default handling via WebserviceRequest
+            switch ($this->wsObject->method) {
+                case 'GET':
+                case 'HEAD':
+                    $this->wsObject->executeEntityGetAndHead();
+                    break;
+                case 'POST':
+                    $this->wsObject->executeEntityPost();
+                    break;
+                case 'PUT':
+                    $this->wsObject->executeEntityPut();
+                    break;
+                case 'DELETE':
+                    $this->wsObject->executeEntityDelete();
+                    break;
+            }
+        }
+        // Need to set an object for the WebserviceOutputBuilder object in any case
+        // because schema need to get webserviceParameters of this object
+        if (isset($object)) {
+            $this->wsObject->objects['empty'] = $object;
+        }
+    }
+
+    /**
+     * Handles attachment file download
+     *
+     * @throws WebserviceException if attachment is not existing or file not available
+     *
+     * @return string[] file details
+     */
+    public function executeFileGetAndHead()
+    {
+        $a = new Attachment((int) $this->wsObject->urlSegment[2]);
+        if (!$a) {
+            throw new WebserviceException(
+                sprintf(
+                    'Attachment %d not found',
+                    $this->wsObject->urlSegment[2]
+                ),
+                [
+                    1,
+                    Response::HTTP_INTERNAL_SERVER_ERROR,
+                ]
+            );
+        }
+
+        // Physical file location
+        $file = _PS_DOWNLOAD_DIR_ . $a->file;
+        // Check if file exists
+        if (!file_exists($file)) {
+            throw new WebserviceException(
+                sprintf(
+                    'Unable to load the attachment file for attachment %d',
+                    $this->wsObject->urlSegment[2]
+                ),
+                [
+                    1,
+                    Response::HTTP_INTERNAL_SERVER_ERROR,
+                ]
+            );
+        }
+
+        // Return file details
+        return array(
+            'file' => $file,
+            'mime' => $a->mime,
+            'file_name' => $a->file_name,
+            'file_size' => $a->file_size,
+        );
+    }
+
+    /**
+     * Handles file upload
+     *
+     * Creates new attachment or replaces existing with a new file.
+     * [PUT] update existing attachment file
+     * [POST] create new attachment
+     */
+    public function executeFileAddAndEdit()
+    {
+        // Load attachment with or without id depending on method
+        $a = $this->wsObject->method == 'PUT' ? new Attachment((int) $this->wsObject->urlSegment[1]) : new Attachment();
```

Comment:
* `$a` should be renamed `$attachment` for instance.
* The ternary test could be limited on the ID affectation, and on another line.